### PR TITLE
Add static value to header propogator

### PR DESCRIPTION
**Pull request by oliviazhang0809**
To support case 
```
{
  "value": "header",
  "to": "specific_path_to_param"
}
```

### CODE REVIEW

File: codegen/header_propagate.go [hunk 1]
```diff
@@ -70,6 +70,17 @@ func (hp *HeaderPropagator) Propagate(
 				val.QualifiedName, field.Name)
 		}
 
+		if val.StaticValue != "" {
```
Comment (cl1337) [inline codegen/header_propagate.go]: can we do type switch here to emphasis 2 different approach (static value vs from field)

Comment (oliviazhang0809) [reply codegen/header_propagate.go]: sure

**Comment by cl1337** [comment]
as discussed offline, all header propagate, header schema, type converting is migrated to our internal project, this case seems general enough, I'm open to either here / where you actually use it

**Comment by ChuntaoLu** [comment]
Setting a default value for a downstream request has no logical relationship with header propagation. This does not seem to be a proper solution for the problem it tries to solve.

**Pull request closed without merging**